Admin/AdminDefintionInterface.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\ItemInterface as MenuItemInterface;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\AdminBundle\Show\ShowMapper;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+interface AdminDefintionInterface
+{
+    /**
+     * Returns the array of allowed export formats.
+     *
+     * @return array
+     */
+    public function getExportFormats();
+
+    /**
+     * @param mixed $object
+     */
+    public function preValidate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function preUpdate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postUpdate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function prePersist($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postPersist($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function preRemove($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postRemove($object);
```

Comment:
These lifecycle methods could be part of yet another interface. Interfaces should be kept small, otherwise no one is going to implement them.